sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -385,7 +428,7 @@ public TokenRequestState(TokenRequestContext currentContext, TaskCompletionSourc
 
                 public bool IsCurrentContextMismatched(TokenRequestContext context) =>
                     (context.Scopes != null && !context.Scopes.AsSpan().SequenceEqual(CurrentContext.Scopes.AsSpan())) ||
-                    (context.Claims != null && !string.Equals(context.Claims, CurrentContext.Claims)) ||
```

Comment:
Why remove the check `context.Claims != null`. This prevents throwing away the cached token on subsequent calls where there is no challenge, no?

Reply:
per offline discussion, I confirmed that always setting `IsCaeEnabled` is safe. This also mitigates the caching concerns.